omod/src/main/java/org/openmrs/module/owa/activator/OwaActivator.java

```diff
@@ -163,4 +172,32 @@ public void stopped() {
 		log.info("OWA Module stopped");
 	}
 	
+	/**
+	 * Used to set the login Url Global Property that directs to the Add-on OWA UI if no UI module
+	 * is installed.
+	 */
+	private void configureLoginUrl(AppManager appManager) {
+		if (appManager.exists(ADD_ON_MANAGER_APP)) {
+			if (!uiModuleInstalled()) {
```

Comment:
What happens when running the reference application without the legacyui module? Or when running any distribution that does not include the legacyui module, but has its own login?

Reply:
Thanks for highlighting this @dkayiwa. Using the legacy module here was a good way to have a working demo. Essentially, we need a better way to inform other modules interested, of the installed UI modules with their own logins. This means having a global property that keeps a list of these installed UI modules, probably in a comma separated string. So modules would have to append their id names to the list through their activators if theirs isn't on the list. In this case, the above check would verify that any of the listed modules is installed.

Reply:
Are you talking about this? https://issues.openmrs.org/browse/TRUNK-5367

Reply:
Thank you @dkayiwa, about TRUNK-5367, the **login.url** GP is always set from the beginning by the core to **login.htm** and that this makes is hard to track whether a UI module is installed, albeit making it also difficult to determine whether a UI module is still installed or has been uninstalled.

My thinking is that where as a login url GP exists to hold the login URL, there is still need to track whether a UI module is installed through a GP of installed UI module.

Reply:
Did you take into consideration the fact that login.url can be overridden by any module?

Reply:
Yes @dkayiwa. The challenge here is, 
1) In which order does the overriding occur and when to/not to override? (Here the last override wins) 
2) What happens when the active UI module is uninstalled from the platform?

The Add On Manager can depend on login.url being empty or containing it's url to confirm there is no UI module installed and this can only happen as long as no UI module has ever been installed. This means only at the beginning of life of the platform before any UI module is installed, will the Add On Manager operate.  

Just my thought, but a million dollar question(maybe not) is how would the overriding capability be designed to work correctly @dkayiwa?